### PR TITLE
Add Cosine, ACF and PACF transformers

### LINKED ISSUE #483: Implement series-to-series transformers
**Describe the solution you'd like**
Here's a simple example of a series-to-series log transformer transformer:

All of these transforms are series-to-series transforms. For an overview of different kinds of transformations, see this [document](https://github.com/sktime/enhancement-proposals/blob/master/steps/03_transformer_api/transform_types.ipynb).

```python
import numpy as np
import pandas as pd

from sktime.transformers.base import _SeriesToSeriesTransformer
from sktime.utils.validation.series import 

class LogTransformer(_SeriesToSeriesTransformer):

    _tags = {"transform-returns-same-time-index": True}

    def transform(self, Z, X=None):
        self.check_is_fitted()
        Z = check_series(Z)
        return np.log(Z)

    def inverse_transform(self, Z, X=None):
        self.check_is_fitted()
        Z = check_series(Z)
        return np.exp(Z)
```

**List of transformations we'd like to include in sktime**
| Transform  | Package/Function |
| ------------- | ------------- |
| FFT/DFT  | https://docs.scipy.org/doc/numpy-1.15.0/reference/routines.fft.html  |
| Autocorrelation function (ACF)  | https://www.statsmodels.org/0.9.0/generated/statsmodels.tsa.stattools.acf.html  |
| Partial autocorrelation function (autoregressive coefficients) (PACF) | https://www.statsmodels.org/0.9.0/generated/statsmodels.tsa.stattools.pacf.html  |
| Auto Regressive Moving Average (ARMA)  | https://www.statsmodels.org/0.9.0/generated/statsmodels.tsa.arima_model.ARMA.html  |
| Power Spectrum  | https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.signal.periodogram.html  |
| Cepstrum  | https://github.com/python-acoustics/python-acoustics/blob/master/acoustics/cepstrum.py  |
| Cosine  | https://docs.scipy.org/doc/numpy/reference/generated/numpy.cos.html  |

**Related issues**
see #6, #224 and #297

### PR DESCRIPTION
**Pull request by Afzal-Ind**
#### Reference Issues/PRs

Fixes #483  

#### What does this implement/fix? Explain your changes.

Added two transformations.

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### PR checklist

- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] I've added the transformers series to the file.
- [x] I've updated the existing file.

### CODE REVIEW

File: sktime/transformers/sktime.transformers.series.acf.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# from statsmodels.tsa.stattools import acf
+from sktime.transformers.base import _SeriesToSeriesTransformer
+from sktime.utils.validation.series import check_series
+
+
+class AutoCorrelationFunctionTransformer(_SeriesToSeriesTransformer):
+    def transform(
+        self,
+        x,
+        unbiased=False,
```
Comment (mloning) [inline sktime/transformers/sktime.transformers.series.acf.py:14]: Make sure you have the same interface here as in the abstract base class. I'd consider the additional arguments here as hyper-parameters and initialise them in `__init__`

File: sktime/transformers/sktime.transformers.series.acf.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        self.unbiased = unbiased
+        self.nlags = nlags
+        self.qstat = qstat
+        self.fft = fft
+        self.alpha = alpha
+        self.missing = missing
+
+        self.check_is_fitted()
+        x = check_series(x)
+        return x
```
Comment (mloning) [inline sktime/transformers/sktime.transformers.series.acf.py:31]: You seem to have forgotten to call the `acf` function

**Review by mloning — COMMENTED** [review]
Hi @Afzal-Ind - thanks for submitting the PR! I had a quick look, the overall structure looks good, I left two comments in the code. 

Next thing to do is to write unit tests for the transformers you added. For an example, check out the [unit tests for the boxcox transformer](https://github.com/alan-turing-institute/sktime/blob/master/sktime/transformers/series/tests/test_boxcox.py)

**Comment by mloning** [comment]
There were some merge conflicts, but I resolved them with the latest commit. To avoid them, you need to make sure to [sync your fork](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork) before you check out a new branch,

**Comment by Afzal-Ind** [comment]
> There were some merge conflicts, but I resolved them with the latest commit.

Hi @mloning. Thanks a lot... I will take care of this.

**Comment by mloning** [comment]
Hi @Afzal-Ind looks much better now - next step is to write some simple unit tests as described above!

**Comment by mloning** [comment]
Hi @Afzal-Ind - okay renaming solved one issue. Now the issue seems to be related to how multivariate data is handled. `SeriesToSeriesTransformer` can take a univariate series (pd.Series) or multivariate series (pd.DataFrame with rows representing time points and columns variables). We need to either (i) enforce univariate data and raise an error if multivariate data is passed by setting `check_series(..., enforce_univariate=True)` or (ii) make sure the transformers correctly handle multivariate data. 

I'd start by setting `check_series(..., enforce_univariate=True)` to get this to run first, we can then add support for multivariate once everything is working.

**Review by Afzal-Ind — COMMENTED** [review]
> Hi @Afzal-Ind - okay renaming solved one issue. Now the issue seems to be related to how multivariate data is handled. `SeriesToSeriesTransformer` can take a univariate series (pd.Series) or multivariate series (pd.DataFrame with rows representing time points and columns variables). We need to either (i) enforce univariate data and raise an error if multivariate data is passed by setting `check_series(..., enforce_univariate=True)` or (ii) make sure the transformers correctly handle multivariate data.
> 
> I'd start by setting `check_series(..., enforce_univariate=True)` to get this to run first, we can then add support for multivariate once everything is working.

Fine now. I can start trying this way. Thanks.

**Comment by mloning** [comment]
Hi @Afzal-Ind if you have time to work on this, we could try to include it in the next release which I'm currently working on. For now, we can just remove the `inverse_transform`.

**Comment by Afzal-Ind** [comment]
Fine @mloning . I am looking forward to working with u on 'inverse_transformer' in next release. Currently dealing with AutoAR. Thanks.

**Comment by mloning** [comment]
@Afzal-Ind I want to push the next release in the next few days, so if you have time to finish it, that would be great. Otherwise we'll include in the next one after that.

**Comment by Afzal-Ind** [comment]
> Of course let me know if you get stuck and I can help!

I was kind of busy for two days. I would start my work on that from tomorrow and let u update.

**Comment by mloning** [comment]
Hi @Afzal-Ind, I've taken a look and made some changes - hopefully all tests pass now. Take a look at [my commit](https://github.com/alan-turing-institute/sktime/pull/509/commits/e987519ce2b194066e07b97c12c204e2fefcc6f3) to understand what I changed - happy to discuss this if you have any questions!

**Pull request merged**